packages/autocomplete-core/src/onInput.ts

```diff
@@ -62,7 +62,7 @@ export function onInput<TItem extends BaseItem>({
   setQuery(query);
   setActiveItemId(props.defaultActiveItemId);
 
-  if (query.length === 0 && props.openOnFocus === false) {
+  if (query && props.openOnFocus === false) {
```

Comment:
this condition is the opposite as before, expected?

Reply:
No, this is why unit tests are failing. Thanks for catching. See 29ca383835c2efea0a4efbdddba329682c467409.